diff --git a/core/modules/ckeditor/js/ckeditor.es6.js b/core/modules/ckeditor/js/ckeditor.es6.js
index fe0ebb940f..96792f2a9b 100644
--- a/core/modules/ckeditor/js/ckeditor.es6.js
+++ b/core/modules/ckeditor/js/ckeditor.es6.js
@@ -382,6 +382,27 @@
       }
     };
   }
+
+  const origBeforeSubmit = Drupal.Ajax.prototype.beforeSubmit;
+  Drupal.Ajax.prototype.beforeSubmit = function (formValues, element, options) {
+    if (typeof CKEDITOR !== 'undefined' && CKEDITOR.instances) {
+      const instances = Object.values(CKEDITOR.instances);
+      instances.forEach((editor) => {
+        formValues.forEach((formField) => {
+          // Get field name from the id in the editor so that it covers all
+          // fields using ckeditor.
+          const element = document.querySelector(`#${editor.name}`);
+          if (element) {
+            const fieldName = element.getAttribute('name');
+            if (formField.name === fieldName && editor.mode === 'source') {
+              formField.value = editor.getData();
+            }
+          }
+        });
+      });
+    }
+    return origBeforeSubmit.apply(this, arguments);
+  };
 })(
   Drupal,
   Drupal.debounce,
diff --git a/core/modules/ckeditor/js/ckeditor.js b/core/modules/ckeditor/js/ckeditor.js
index be798f0e05..9f8db6d9b7 100644
--- a/core/modules/ckeditor/js/ckeditor.js
+++ b/core/modules/ckeditor/js/ckeditor.js
@@ -202,4 +202,27 @@
       }
     };
   }
+
+  var origBeforeSubmit = Drupal.Ajax.prototype.beforeSubmit;
+
+  Drupal.Ajax.prototype.beforeSubmit = function (formValues, element, options) {
+    if (typeof CKEDITOR !== 'undefined' && CKEDITOR.instances) {
+      var instances = Object.values(CKEDITOR.instances);
+      instances.forEach(function (editor) {
+        formValues.forEach(function (formField) {
+          var element = document.querySelector("#".concat(editor.name));
+
+          if (element) {
+            var fieldName = element.getAttribute('name');
+
+            if (formField.name === fieldName && editor.mode === 'source') {
+              formField.value = editor.getData();
+            }
+          }
+        });
+      });
+    }
+
+    return origBeforeSubmit.apply(this, arguments);
+  };
 })(Drupal, Drupal.debounce, CKEDITOR, jQuery, Drupal.displace, Drupal.AjaxCommands);
\ No newline at end of file
