Index: web/modules/contrib/localgov_events/localgov_events.module
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web/modules/contrib/localgov_events/localgov_events.module	(revision f95d1c239f82d0a971ca3a02f0a0bcbc8866c701)
+++ web/modules/contrib/localgov_events/localgov_events.module	(date 1662485881890)
@@ -243,19 +243,23 @@
  * Implements hook_views_query_alter().
  */
 function localgov_events_views_pre_view(ViewExecutable $view) {
-  if ($view->id() == 'localgov_events_listing') {
-
-    // Add one day to end date so events for that day are included.
-    $filters = $view->getExposedInput();
-    if (isset($filters['end']) && strlen($filters['end']) > 0) {
-      $filters['end'] = date('Y-m-d', strtotime($filters['end'] . ' + 1 days'));
-      $view->setExposedInput($filters);
-    }
+  // We check if the custom localgov module that overwrites required parts of the localgov_events_listing view is
+  // installed, if so we dont run any contrib code from here.
+  $custom_module = 'bc_localgov_events';
+  if (!\Drupal::moduleHandler()->moduleExists($custom_module)) {
+    if ($view->id() == 'localgov_events_listing') {
+      // Add one day to end date so events for that day are included.
+      $filters = $view->getExposedInput();
+      if (isset($filters['end']) && strlen($filters['end']) > 0) {
+        $filters['end'] = date('Y-m-d', strtotime($filters['end'] . ' + 1 days'));
+        $view->setExposedInput($filters);
+      }
 
-    // Set default to todays date if empty.
-    if (empty($filters['start'])) {
-      $filters['start'] = date('Y-m-d');
-      $view->setExposedInput($filters);
+      // Set default to todays date if empty.
+      if (empty($filters['start'])) {
+        $filters['start'] = date('Y-m-d');
+        $view->setExposedInput($filters);
+      }
     }
   }
 }
